client/src/HtmlLtacProf.ts

```diff
@@ -55,12 +55,14 @@ export class HtmlLtacProf {
     }
 
     const httpServer = (this.httpServer = http.createServer());
-    this.serverReady = new Promise<void>((resolve, reject) =>
-      httpServer.listen(0, "localhost", undefined, (e: any) => {
-        if (e) reject(e);
-        else resolve();
-      })
-    );
+    this.serverReady = new Promise<void>((resolve, reject) => {
+      httpServer.listen(0);
+      httpServer.on("error", (e: Error) => {
+        console.log(e);
+        reject(e);
```

Comment:
This also looks incorrect: `resolve` will be called before any error event occurs, most likely, even if starting the server fails.

Reply:
I'll defer to your judgement on this, because I honestly didn't know what I was doing when I was editing this file: I don't know how Promises work.